### PR TITLE
Add SchedulerCallableSupportHPO

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

- Ticket no. 137956
- As same as `OptimizerCallableSupportHPO` added in #3258, the scheduler callable should also be picklable and allow access on its parameters.

### How to test
Added unit tests for this change as well.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/schedulers/callable.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+from torch.optim.lr_scheduler import LRScheduler
+from torch.optim.lr_scheduler import ReduceLROnPlateau as TorchReduceLROnPlateau
+
+from otx.core.utils.jsonargparse import ClassType, lazy_instance
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable
+
+
+class SchedulerCallableSupportHPO:
```
Comment (harimkang) [inline src/otx/core/schedulers/callable.py:23]: It's clear that this is a class to support HPO, but in the actual code it appears to be a Callable for Pickable.  Wouldn't the existing `PicklableLRSchedulerCallable` be a better name?

**Review by harimkang — APPROVED** [review]
LGTM, I left one minor comment, but you can ignore it.

File: src/otx/core/schedulers/callable.py (hunk 1 above)
Comment (vinnamkim) [reply src/otx/core/schedulers/callable.py:23]: > the scheduler callable should also be picklable and allow access on its parameters.

It should satisfy both aspects: 1) picklable and 2) param accessible

File: src/otx/core/schedulers/callable.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        scheduler = func(optimizer)
+
+        allow_names = set(inspect.signature(scheduler.__class__).parameters)
+
+        if isinstance(scheduler, ReduceLROnPlateau):
+            # NOTE: Other arguments except "monitor", such as "patience"
+            # are not included in the signature of ReduceLROnPlateau.__init__()
+            allow_names.update(key for key in inspect.signature(TorchReduceLROnPlateau).parameters)
+
+        block_names = {"optimizer", "last_epoch"}
```
Comment (eunwoosh) [inline src/otx/core/schedulers/callable.py:151]: Why is last_epoch excluded?

**Review by eunwoosh — APPROVED** [review]
Thanks for your work! I'll make following PR for HPO. And I left one minor question. Please take a look.

File: src/otx/core/schedulers/callable.py (hunk 2 above)
Comment (vinnamkim) [reply src/otx/core/schedulers/callable.py:151]: image
`scheduler.state_dict()` returns `last_epoch=0` even if it gets `last_epoch=-1` (default) init argument. `last_epoch>=0` has a problem if not resuming an optimizer.
